### ISSUE TITLE
support for selected test run/debug in docker (or at least venv)

### ISSUE THREAD

**Issue by dz0** [trinhanhngoc/pycharm-odoo#100]
Hi, 

I know this is not the functionality you currently target, but you might know context :)

I am struggling to setup tests running/debugg'ing   via https://www.jetbrains.com/help/pycharm/using-docker-as-a-remote-interpreter.html#run

maybe you have some hints or suggestions...

for example, for Django pycharm has some helpers, maybe they can be applied to odoo?

ps.: also is it possible to setup `Python Console` to be bound to docker env

**Comment by trinhanhngoc** [comment]
Hi @dz0 ,

What version of PyCharm are you using?

**Comment by dz0** [comment]
> What version of PyCharm are you using?

PyCharm 2022.1.1 (Professional Edition)
Build #PY-221.5591.52, built on May 10, 2022

on Ubuntu 20.04

**Comment by dz0** [comment]
maybe found  a  non-docker way (haven't tried)
https://github.com/camptocamp/pytest-odoo/issues/31#issuecomment-489579074

**Comment by trinhanhngoc** [comment]
@dz0 

I tried running/debugging Odoo with Docker in PyCharm 2022.2 but due to some regression issues in PyCharm 2022.2, I could not do it. I will try again after JetBrains releases PyCharm 2022.2.1. Anyway, what problem do you have with **venv**?

**Comment by dz0** [comment]
(how) does it work now?

**Comment by trinhanhngoc** [comment]
Hello @dz0 ,

Sorry if I misunderstood your problem again. Could you provide some screenshots to demonstrate the problem?

**Comment by dz0** [comment]
How can I run Odoo-ide based tests for odoo instance in the docker?

I guess it could 
- 1) pass command
- 2) parse the output (without needing to controll remote interpreter directly)

**Comment by trinhanhngoc** [comment]
@dz0 ,

You mean running odoo in docker without [configuring an interpreter using Docker](https://www.jetbrains.com/help/pycharm/using-docker-as-a-remote-interpreter.html)?

**Comment by dz0** [comment]
1) yes, would be nice -- via some `docker exec`.

2) also, it is ok, if it would work via configuration you linked... -- if setting the interpreter as remote is enough :)
just for now (for  3 years :) I am running locally, as probably didn't manage to setup sth properly... especially for debug (but that is another story).

PS: Maybe the issue can be in tracebacks non-matchig paths locally and on docker -- but symlinking can help ;)

**Comment by trinhanhngoc** [comment]
> 1. yes, would be nice -- via some `docker exec`.
> 2. also, it is ok, if it would work via configuration you linked... -- if setting the interpreter as remote is enough :)
>    just for now (for  3 years :) I am running locally, as probably didn't manage to setup sth properly... especially for debug (but that is another story).
> 
> PS: Maybe the issue can be in tracebacks non-matchig paths locally and on docker -- but symlinking can help ;)

Yes, The Odoo run configuration should work if you configure an interpreter using Docker.

<img width="3552" height="1998" alt="Image" src="https://github.com/user-attachments/assets/131e0a8c-7919-448b-96c3-592e6069bcdb" />
